src/localization/I18nProvider.tsx

```diff
@@ -39,11 +39,16 @@ const I18nProvider = ({ children }: Props) => {
       return;
     }
 
-    loadCatalog(locale).then(({ messages }) => {
-      i18n.load(locale, messages);
-      i18n.activate(locale);
-      setActiveLocale(locale);
-    });
+    loadCatalog(locale)
+      .then(({ messages }) => {
+        i18n.load(locale, messages);
+        i18n.activate(locale);
+        setActiveLocale(locale);
+      })
+      .catch((error) => {
```

Comment:
This is currently only executed on the client side, do the rejection handling of node should not be relevant making this catch unnecessary. I think we can maintain the old behavior.

Reply:
I believe that this comes up in tests. Going to pull the branch and check but currently when running tests there are (sometimes many) uncaught i18n-related throws.

Reply:
this does affect tests

Reply:
I would need to figure out how to mock enough objects to where the promise would succeed. For now, I wanted to get the render test in `src/Root.test.tsx` to pass. In my research of the error message generated when upgrading from node 14.x to node 16.x I found [this article](https://betterprogramming.pub/whats-new-in-node-js-15-fc24e87e2590) which suggested that the affected change in warning threshold was added in node 15. 

In reviewing this file, I did notice that the extra return statement inside the catch is not needed as the other statements successfully cover the branching with a return statement that either creates the component or returns null anyway.